### ISSUE TITLE
Allow to configure the prefix path

### ISSUE THREAD

**Issue by edupo** [randombenj/docat#80]
I tried to configure docat after my other Nginx server where Jenkins is running. The root path of the server is reserved also.

It would be great to allow to set somewhere the prefix path of the app so it would be possible to use it from a different path:
```
myserver/docat/#
myserver/docat/doc/myproject/latest
```

The backend works without issue, but the web interface has troubles dealing with that. You can enable part of the frontend by rebuilding the container while modifying the `vue.config.js` to contain `publicPath: '/docat/'` but that does not make it entirely work.

**Comment by palto42** [comment]
For my local environment I can't use a dedicated sub-domain for docat, so I would like to reverse-proxy it on a sub-path. Adding such prefix path option would be really great!

**Comment by reglim** [comment]
@edupo @palto42 

I tried doing this, but I didn't manage to get it to work either.

What I tried:

1. Add basename to the hashrouter:
     This didn't work because the uri would then be `/#/docat/...`
2. Switch to browser router and add basename there:
     This didn't work because it wouldn't recognize the hashtag we have in the url for backwards-compatability:
     ``` 
          <Router basename="/test/#"> is not able to match the URL "/test/#/" because it does not start with the basename, so the <Router> won't render anything.
     ``` 
3. I also tried replacing the hashtag with it's url encoded counterpart, but that didn't work either, because it didn't interpret the url encoded hashtag as a hashtag
4. Finally I tried adding it in the nginx config, but I couldn't get it to work there either and it wouldn't help you locally anyway.

If anyone has another idea on how to fix that, you're welcome to try.

Also here are the sources I followed:
https://www.kindacode.com/article/how-to-change-the-base-path-in-react/ 
https://serverfault.com/questions/853559/nginx-place-a-prefix-in-my-url
https://stackoverflow.com/questions/70519315/router-basename-is-not-able-to-match-the-url-because-it-does-not-start-with-the
https://github.com/remix-run/react-router/pull/8460#issuecomment-990415343
https://github.com/remix-run/react-router/issues/8459

**Comment by palto42** [comment]
@reglim thanks for your effort to try and find a solution.
Unfortunately I have no experience with react, so can't help much.

I found this page about sub-site topic, but not sure if it provides any new ideas: https://muffinman.io/blog/react-router-subfolder-on-server/

What exactly did you try in #4 with nginx config? The above page suggests that the HashRouter should work (without specifying `basename`) if the app is deployed in a subfolder of the web server.

**Comment by reglim** [comment]
@palto42 

Thanks for the article, I'll try that out and get back to you.

I tried something like this, though I'm not sure that this is even remotely correct:

```
    location /docat/doc {
        root /var/docat;
    }

    location /docat/api {
        client_max_body_size 100M;
        proxy_pass http://python_backend;
    }

    location /docat {
    }
```

**Comment by palto42** [comment]
@reglim Thanks, this is what I also tried for the first option in the article I shared, but didn't work as expected.

**Comment by reglim** [comment]
@palto42 @edupo

Ok, so I managed to get it to work in using a build arg for docat. The exact instructions are in the [ReadMe](https://github.com/docat-org/docat/tree/improvement/80-prefix-path). 

I'd be glad if you could try it out and get back to me to tell me whether or not it works properly. 
It's a bit janky because we need to rebuild the whole web project, replace the paths in the nginx config and include it in the API.

**Comment by palto42** [comment]
@reglim Thanks for the update, tried it with a docker image built from the new branch as per instructions, but I now get a blank page under the prefix path and in the page source I see an error "We're sorry docat web doesn't work properly without JavaScript enabled. Please enable it to continue."
My browser has javascript enabled, so not sure what causes this error.

**Comment by reglim** [comment]
@palto42 

Yes, I tried including the Prefix Path in the `package.json`, but somehow `envsubst` didn't work. Anyway, including that would kind of break the compatability with local development, as you would have to have the configuration with an environment variable in there:

```json
"homepage": "$PREFIX_PATH",
```

I think that this is a rather niche usecase compared to local development, so I think no prefix path is not such a bad default.

Also, I think that tagging the docker image is a good idea. I'll add it to the ReadMe : )

**Comment by palto42** [comment]
@rglim Thanks, I understand your rationale. I will try to avoid prefix path, forking will be too much effort.
